opensourceleg/actuators/dephy.py

```diff
@@ -0,0 +1,1739 @@
+        """
+        Sets the position gains in arbitrary Dephy units.
+
+        Args:
+            kp (float): The proportional gain
+            ki (float): The integral gain
+            kd (float): The derivative gain
+            ff (float): The feedforward gain
+        """
+        self.set_gains(
+            kp=int(kp),
+            ki=int(ki),
+            kd=int(kd),
+            k=0,
+            b=0,
+            ff=int(ff),
+        )
+
+    def set_current_gains(
+        self,
+        kp: float = DEFAULT_CURRENT_GAINS.kp,
+        ki: float = DEFAULT_CURRENT_GAINS.ki,
+        kd: float = DEFAULT_CURRENT_GAINS.kd,
+        ff: float = DEFAULT_CURRENT_GAINS.ff,
+    ) -> None:
+        """
+        Sets the current gains in arbitrary Dephy units.
+
+        Args:
+            kp (float): The proportional gain
+            ki (float): The integral gain
+            kd (float): The derivative gain
+            ff (float): The feedforward gain
+        """
+        self.set_gains(
+            kp=int(kp),
+            ki=int(ki),
+            kd=int(kd),
+            k=0,
+            b=0,
+            ff=int(ff),
+        )
+
+    def set_motor_impedance(
+        self,
+        kp: float = DEFAULT_IMPEDANCE_GAINS.kp,
+        ki: float = DEFAULT_IMPEDANCE_GAINS.ki,
+        kd: float = DEFAULT_IMPEDANCE_GAINS.kd,
+        k: float = 0.08922,
+        b: float = 0.0038070,
+        ff: float = 128,
+    ) -> None:
+        """
+        Set the impedance gains of the motor in real units: Nm/rad and Nm/rad/s.
+
+        Args:
+            kp (float): Proportional gain. Defaults to 40.
+            ki (float): Integral gain. Defaults to 400.
+            kd (float): Derivative gain. Defaults to 0.
+            k (float): Spring constant. Defaults to 0.08922 Nm/rad.
+            b (float): Damping constant. Defaults to 0.0038070 Nm/rad/s.
+            ff (float): Feedforward gain. Defaults to 128.
+        """
+        self.set_impedance_gains(
+            kp=kp,
+            ki=ki,
+            kd=kd,
+            k=int(k * self.MOTOR_CONSTANTS.NM_PER_RAD_TO_K),
+            b=int(b * self.MOTOR_CONSTANTS.NM_S_PER_RAD_TO_B),
+            ff=ff,
+        )
+
+    def set_joint_impedance(
+        self,
+        kp: float = DEFAULT_IMPEDANCE_GAINS.kp,
+        ki: float = DEFAULT_IMPEDANCE_GAINS.ki,
+        kd: float = DEFAULT_IMPEDANCE_GAINS.kd,
+        k: float = 100.0,
+        b: float = 3.0,
+        ff: float = 128,
+    ) -> None:
+        """
+        Set the impedance gains of the joint in real units: Nm/rad and Nm/rad/s.
+        This sets the impedance at the output and automatically scales based on gear raitos.
+
+        Conversion:
+            K_motor = K_joint / (gear_ratio ** 2)
+            B_motor = B_joint / (gear_ratio ** 2)
+
+        Args:
+            kp (float): Proportional gain. Defaults to 40.
+            ki (float): Integral gain. Defaults to 400.
+            kd (float): Derivative gain. Defaults to 0.
```

Comment:
We can be more descriptive here, as this is a bit confusing. The `kp`, `ki`, `kd` here are all related to the current controller beneath the impedance controller right? Perhaps we detail that in the docstring. Also I don't know what `kd` does in this context. If it does nothing, we shouldn't include it as an input.